ncrypt/crypt.c

```diff
@@ -1114,15 +1124,86 @@ int mutt_protected_headers_handler(struct Body *b_email, struct State *state)
   const bool c_weed = cs_subset_bool(NeoMutt->sub, "weed");
   const short c_wrap = cs_subset_number(NeoMutt->sub, "wrap");
   const int wraplen = display ? mutt_window_wrap_cols(state->wraplen, c_wrap) : 0;
+  const CopyHeaderFlags chflags = display ? CH_DISPLAY : CH_NO_FLAGS;
+  struct Buffer *buf = buf_pool_get();
+
+  if (b_email->mime_headers->date &&
+      (!display || !c_weed || !mutt_matches_ignore("date")))
+  {
+    mutt_write_one_header(state->fp_out, "Date", b_email->mime_headers->date,
+                          state->prefix, wraplen, chflags, NeoMutt->sub);
+  }
+
+  if (!display || !c_weed || !mutt_matches_ignore("return-path")) {
+    mutt_addrlist_write(&b_email->mime_headers->return_path, buf, display);
+    mutt_write_one_header(state->fp_out, "Return-Path", buf_string(buf),
+                          state->prefix, wraplen, chflags, NeoMutt->sub);
+  }
```

Comment:
Please can you `clang-format` all the `{}`s

Reply:
I never used that program.  If you tell me the arguments you pass to it, I can check it.

Reply:
It would be interesting to have `make lint` to run it, BTW.

Reply:
Ahhhh, it was easy!

```diff
$ diff -u <(clang-format ncrypt/crypt.c) ncrypt/crypt.c
--- /dev/fd/63	2024-04-29 18:54:01.607735034 +0200
+++ ncrypt/crypt.c	2024-04-29 14:15:10.709854184 +0200
@@ -1127,62 +1127,55 @@
   const CopyHeaderFlags chflags = display ? CH_DISPLAY : CH_NO_FLAGS;
   struct Buffer *buf = buf_pool_get();
 
-  if (b_email->mime_headers->date && (!display || !c_weed || !mutt_matches_ignore("date")))
+  if (b_email->mime_headers->date &&
+      (!display || !c_weed || !mutt_matches_ignore("date")))
   {
     mutt_write_one_header(state->fp_out, "Date", b_email->mime_headers->date,
                           state->prefix, wraplen, chflags, NeoMutt->sub);
   }
 
-  if (!display || !c_weed || !mutt_matches_ignore("return-path"))
-  {
+  if (!display || !c_weed || !mutt_matches_ignore("return-path")) {
     mutt_addrlist_write(&b_email->mime_headers->return_path, buf, display);
     mutt_write_one_header(state->fp_out, "Return-Path", buf_string(buf),
                           state->prefix, wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("from"))
-  {
+  if (!display || !c_weed || !mutt_matches_ignore("from")) {
     buf_reset(buf);
     mutt_addrlist_write(&b_email->mime_headers->from, buf, display);
-    mutt_write_one_header(state->fp_out, "From", buf_string(buf), state->prefix,
-                          wraplen, chflags, NeoMutt->sub);
+    mutt_write_one_header(state->fp_out, "From", buf_string(buf),
+                          state->prefix, wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("to"))
-  {
+  if (!display || !c_weed || !mutt_matches_ignore("to")) {
     buf_reset(buf);
     mutt_addrlist_write(&b_email->mime_headers->to, buf, display);
-    mutt_write_one_header(state->fp_out, "To", buf_string(buf), state->prefix,
-                          wraplen, chflags, NeoMutt->sub);
+    mutt_write_one_header(state->fp_out, "To", buf_string(buf),
+                          state->prefix, wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("cc"))
-  {
+  if (!display || !c_weed || !mutt_matches_ignore("cc")) {
     buf_reset(buf);
     mutt_addrlist_write(&b_email->mime_headers->cc, buf, display);
-    mutt_write_one_header(state->fp_out, "Cc", buf_string(buf), state->prefix,
-                          wraplen, chflags, NeoMutt->sub);
+    mutt_write_one_header(state->fp_out, "Cc", buf_string(buf),
+                          state->prefix, wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("sender"))
-  {
+  if (!display || !c_weed || !mutt_matches_ignore("sender")) {
     buf_reset(buf);
     mutt_addrlist_write(&b_email->mime_headers->sender, buf, display);
     mutt_write_one_header(state->fp_out, "Sender", buf_string(buf),
                           state->prefix, wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("reply-to"))
-  {
+  if (!display || !c_weed || !mutt_matches_ignore("reply-to")) {
     buf_reset(buf);
     mutt_addrlist_write(&b_email->mime_headers->reply_to, buf, display);
     mutt_write_one_header(state->fp_out, "Reply-To", buf_string(buf),
                           state->prefix, wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("mail-followup-to"))
-  {
+  if (!display || !c_weed || !mutt_matches_ignore("mail-followup-to")) {
     buf_reset(buf);
     mutt_addrlist_write(&b_email->mime_headers->mail_followup_to, buf, display);
     mutt_write_one_header(state->fp_out, "Mail-Followup-To", buf_string(buf),
                           state->prefix, wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("x-original-to"))
-  {
+  if (!display || !c_weed || !mutt_matches_ignore("x-original-to")) {
     buf_reset(buf);
     mutt_addrlist_write(&b_email->mime_headers->x_original_to, buf, display);
     mutt_write_one_header(state->fp_out, "X-Original-To", buf_string(buf),
@@ -1196,15 +1189,13 @@
                           state->prefix, wraplen, chflags, NeoMutt->sub);
   }
 
-  if (!display || !c_weed || !mutt_matches_ignore("references"))
-  {
+  if (!display || !c_weed || !mutt_matches_ignore("references")) {
     buf_reset(buf);
     mutt_list_write(&b_email->mime_headers->references, buf);
     mutt_write_one_header(state->fp_out, "References", buf_string(buf),
                           state->prefix, wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("in-reply-to"))
-  {
+  if (!display || !c_weed || !mutt_matches_ignore("in-reply-to")) {
     buf_reset(buf);
     mutt_list_write(&b_email->mime_headers->in_reply_to, buf);
     mutt_write_one_header(state->fp_out, "In-Reply-To", buf_string(buf),
```

Reply:
Hmmm, it's very unclear to me when the brace goes in a new line and when it is in the same line.  I thought you put it always in a new line, as in

```c
$ grepc disable_coredumps .
./ncrypt/crypt.c:static void disable_coredumps(void)
{
  struct rlimit rl = { 0, 0 };
  static bool done = false;

  if (!done)
  {
    setrlimit(RLIMIT_CORE, &rl);
    done = true;
  }
}
```

But clang-format is suggesting the same line for my changes.  What's the pattern?

Reply:
```yaml
$ grep -i brace .clang-format 
BreakBeforeBraces: Allman
Cpp11BracedListStyle:             false
```

Hmmm.  I don't see why it's suggesting to move the braces.  :S

Reply:
Ahhh, no, I was looking at the diff the other way around!

```diff
$ diff -u ncrypt/crypt.c <(clang-format ncrypt/crypt.c)
--- ncrypt/crypt.c	2024-04-29 14:15:10.709854184 +0200
+++ /dev/fd/63	2024-04-29 19:02:04.986880689 +0200
@@ -1127,55 +1127,62 @@
   const CopyHeaderFlags chflags = display ? CH_DISPLAY : CH_NO_FLAGS;
   struct Buffer *buf = buf_pool_get();
 
-  if (b_email->mime_headers->date &&
-      (!display || !c_weed || !mutt_matches_ignore("date")))
+  if (b_email->mime_headers->date && (!display || !c_weed || !mutt_matches_ignore("date")))
   {
     mutt_write_one_header(state->fp_out, "Date", b_email->mime_headers->date,
                           state->prefix, wraplen, chflags, NeoMutt->sub);
   }
 
-  if (!display || !c_weed || !mutt_matches_ignore("return-path")) {
+  if (!display || !c_weed || !mutt_matches_ignore("return-path"))
+  {
     mutt_addrlist_write(&b_email->mime_headers->return_path, buf, display);
     mutt_write_one_header(state->fp_out, "Return-Path", buf_string(buf),
                           state->prefix, wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("from")) {
+  if (!display || !c_weed || !mutt_matches_ignore("from"))
+  {
     buf_reset(buf);
     mutt_addrlist_write(&b_email->mime_headers->from, buf, display);
-    mutt_write_one_header(state->fp_out, "From", buf_string(buf),
-                          state->prefix, wraplen, chflags, NeoMutt->sub);
+    mutt_write_one_header(state->fp_out, "From", buf_string(buf), state->prefix,
+                          wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("to")) {
+  if (!display || !c_weed || !mutt_matches_ignore("to"))
+  {
     buf_reset(buf);
     mutt_addrlist_write(&b_email->mime_headers->to, buf, display);
-    mutt_write_one_header(state->fp_out, "To", buf_string(buf),
-                          state->prefix, wraplen, chflags, NeoMutt->sub);
+    mutt_write_one_header(state->fp_out, "To", buf_string(buf), state->prefix,
+                          wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("cc")) {
+  if (!display || !c_weed || !mutt_matches_ignore("cc"))
+  {
     buf_reset(buf);
     mutt_addrlist_write(&b_email->mime_headers->cc, buf, display);
-    mutt_write_one_header(state->fp_out, "Cc", buf_string(buf),
-                          state->prefix, wraplen, chflags, NeoMutt->sub);
+    mutt_write_one_header(state->fp_out, "Cc", buf_string(buf), state->prefix,
+                          wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("sender")) {
+  if (!display || !c_weed || !mutt_matches_ignore("sender"))
+  {
     buf_reset(buf);
     mutt_addrlist_write(&b_email->mime_headers->sender, buf, display);
     mutt_write_one_header(state->fp_out, "Sender", buf_string(buf),
                           state->prefix, wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("reply-to")) {
+  if (!display || !c_weed || !mutt_matches_ignore("reply-to"))
+  {
     buf_reset(buf);
     mutt_addrlist_write(&b_email->mime_headers->reply_to, buf, display);
     mutt_write_one_header(state->fp_out, "Reply-To", buf_string(buf),
                           state->prefix, wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("mail-followup-to")) {
+  if (!display || !c_weed || !mutt_matches_ignore("mail-followup-to"))
+  {
     buf_reset(buf);
     mutt_addrlist_write(&b_email->mime_headers->mail_followup_to, buf, display);
     mutt_write_one_header(state->fp_out, "Mail-Followup-To", buf_string(buf),
                           state->prefix, wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("x-original-to")) {
+  if (!display || !c_weed || !mutt_matches_ignore("x-original-to"))
+  {
     buf_reset(buf);
     mutt_addrlist_write(&b_email->mime_headers->x_original_to, buf, display);
     mutt_write_one_header(state->fp_out, "X-Original-To", buf_string(buf),
@@ -1189,13 +1196,15 @@
                           state->prefix, wraplen, chflags, NeoMutt->sub);
   }
 
-  if (!display || !c_weed || !mutt_matches_ignore("references")) {
+  if (!display || !c_weed || !mutt_matches_ignore("references"))
+  {
     buf_reset(buf);
     mutt_list_write(&b_email->mime_headers->references, buf);
     mutt_write_one_header(state->fp_out, "References", buf_string(buf),
                           state->prefix, wraplen, chflags, NeoMutt->sub);
   }
-  if (!display || !c_weed || !mutt_matches_ignore("in-reply-to")) {
+  if (!display || !c_weed || !mutt_matches_ignore("in-reply-to"))
+  {
     buf_reset(buf);
     mutt_list_write(&b_email->mime_headers->in_reply_to, buf);
     mutt_write_one_header(state->fp_out, "In-Reply-To", buf_string(buf),
```

Now it's clear.  :-)

Reply:
> It would be interesting to have `make lint` to run it on all files, BTW.

Yes... but there's a slight problem.
Different versions on `clang-format` never seem to produce exactly the same results.
I don't want to get into a formatting war :-)

I have an alias to limit `clang-format` to just those _changed_ files.

```sh
# clang-format the diff
alias cfd="git diff HEAD --name-only -- '*.c' | xargs --no-run-if-empty clang-format -i"
```